clerk/jwks_cache.go

```diff
@@ -0,0 +1,32 @@
+package clerk
+
+import (
+	"fmt"
+	"time"
+
+	"gopkg.in/square/go-jose.v2"
+)
+
+type jwksCache struct {
+	jwks      *JWKS
+	expiresAt time.Time
+}
+
+func (c jwksCache) isInvalid() bool {
+	return c.jwks == nil || len(c.jwks.Keys) == 0 || time.Now().After(c.expiresAt)
+}
+
+func (c *jwksCache) set(jwks *JWKS) {
+	c.jwks = jwks
+	c.expiresAt = time.Now().Add(time.Hour)
+}
+
+func (c jwksCache) get(kid string) (*jose.JSONWebKey, error) {
+	for _, key := range c.jwks.Keys {
+		if key.KeyID == kid {
```

Comment:
🔧 We should raise an error if there is no match to fail fast.

Reply:
We have to iterate first over the keys to check if there is a match with the provided kid